core/rawdb/schema.go

```diff
@@ -152,6 +152,11 @@ func headerKey(number uint64, hash common.Hash) []byte {
 	return append(append(headerPrefix, encodeBlockNumber(number)...), hash.Bytes()...)
 }
 
+// extHeaderKey = headerPrefix + num (uint64 big endian) + location + context + hash
+func extHeaderKey(number uint64, context uint64, hash common.Hash) []byte {
+	return append(append(append(headerPrefix, encodeBlockNumber(number)...), encodeBlockNumber(context)...), hash.Bytes()...)
```

Comment:
Is it worth adding an `encodeContext(...)` utility? I imagine there are other places we need to encode context

Reply:
So for extHeaderKey we should be using location and not context.

For block bodies we will need to use both location and context.